File: live-coding/go-source/session1/go-basic/package2/my_structs.go
package package2

import "fmt"

type Species interface {
	Describe() string
}

type Human struct {
	name   string
	age    int
	wealth *int
}

func (h Human) Describe() string {
	return fmt.Sprintf("I am a human named %s and I am %d years old", h.name, h.age)
}

type Animal struct {
	Name string
	Age  int
}

func (a Animal) Describe() string {
	return fmt.Sprintf("I am an animal named %s and I am %d years old", a.Name, a.Age)
}

func (h *Human) SetName(name string) {
	h.name = name
}

func (h *Human) SetAge(age int) {
	h.age = age
}

func (h *Human) SetWealth(wealth *int) {
	h.wealth = wealth
}

func (h Human) String() string {
	return fmt.Sprintf("Name: %s, Age: %d", h.name, h.age)
}

func (h Human) IncorrectSetAge(age int) {
	h.age = age
}

func (h Human) IncorrectSetName(name string) {
	h.name = name
}

File: live-coding/go-source/session1/py-basic/my_structs.py
class Species:
    def Describe(self):
        print("I belong to species")

class Cat(Species):
    def __init__(self, name):
        self.name = name

class Human(Species):
    def __init__(self, name, age):
        self.name = name
        self.age = age
    
    def set_name(self, name):
        self.name = name

    def set_age(self, age):
        self.age = age

    def __str__(self):
        return f"Name: {self.name}, Age: {self.age}"
    def Describe(self):
        print( "I am a human ", self)
    



if __name__ == "__main__":
    cat=Cat("Minouch")
    cat.Describe()
    
    human = Human("Geir", 30)
    #print(human)

    #human.set_name("Geir Arne")
    #human.set_age(31)

    #print(human)

    human.Describe()